File: PDOO-Civitas/CivitasJ/src/GUI/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package GUI;

import civitas.CivitasJuego;
import civitas.GestionesInmobiliarias;
import civitas.Jugador;
import civitas.OperacionInmobiliaria;
import civitas.Operaciones_juego;
import civitas.SalidasCarcel;
import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 */
public class Controlador {
    
    private CivitasJuego juego;
    private CivitasView vista;
    
    public Controlador(CivitasJuego juego, CivitasView vista){
    
        this.juego = juego;
        this.vista = vista;
    }
    
    public void juega(){
    
        vista.setCivitasJuego(juego);
        
        while(!juego.finalDelJuego()){
        
            vista.actualizarVista();
            
            Operaciones_juego operacion = juego.siguientePaso();
            
            vista.mostrarSiguienteOperacion(operacion);
            
            if(operacion != Operaciones_juego.PASAR_TURNO){
            
                vista.mostrarEventos();
            }
            
            if(!juego.finalDelJuego()){
            
                if(operacion == Operaciones_juego.COMPRAR){
                
                    Respuestas respuesta = vista.comprar();
                    
                    if(respuesta == Respuestas.SI){
                    
                        juego.comprar();
                    }
                    
                    juego.siguientePasoCompletado(Operaciones_juego.COMPRAR);
                }
                
                if(operacion == Operaciones_juego.GESTIONAR){
                
                    vista.gestionar();
                    
                    int ig = vista.getGestion();
                    int ip = vista.getPropiedad();
                    
                    OperacionInmobiliaria oi = new OperacionInmobiliaria(GestionesInmobiliarias.values()[ig], ip);
                    
                    if(oi.getGestion() == GestionesInmobiliarias.CANCELAR_HIPOTECA){
                    
                        juego.cancelarHipoteca(ip);
                    }
                    else if(oi.getGestion() == GestionesInmobiliarias.CONSTRUIR_CASA){
                    
                        juego.construirCasa(ip);
                    }
                    else if(oi.getGestion() == GestionesInmobiliarias.CONSTRUIR_HOTEL){
                    
                        juego.construirHotel(ip);
                    }
                    else if(oi.getGestion() == GestionesInmobiliarias.HIPOTECAR){

                        juego.hipotecar(ip);
                    }
                    else if(oi.getGestion() == GestionesInmobiliarias.VENDER){
                    
                        juego.vender(ip);
                    }
                    else{
                    
                        juego.siguientePasoCompletado(operacion);
                    }
                }
                
                if(operacion == Operaciones_juego.SALIR_CARCEL){
                
                    SalidasCarcel salida = vista.salirCarcel();
                    
                    if(salida == SalidasCarcel.PAGANDO){
                    
                        juego.salirCarcelPagando();
                    }
                    else{
                    
                        juego.salirCarcelTirando();
                    }
                    
                    juego.siguientePasoCompletado(operacion);
                }
            }
            
            vista.actualizarVista();
        }
        
    }
}

File: PDOO-Civitas/CivitasR/lib/controlador.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "civitas_juego"
require_relative "gestiones_inmobiliarias"
require_relative "operacion_inmobiliaria"
require_relative "operaciones_juego"
require_relative "respuestas"
require_relative "salidas_carcel"

module Civitas
  class Controlador
    
    def initialize(j, v)
      
      @juego = j
      @vista = v
      
    end
    
    def juega()
      
      @vista.setCivitasJuego(@juego)
      
      while(!@juego.final_del_juego())
        
        @vista.actualizarVista()
        
        @vista.pausa()
        
        operacion = @juego.siguiente_paso()
        
        @vista.mostrarSiguienteOperacion(operacion)
        
        if(operacion != Operaciones_juego::PASAR_TURNO)
          
          @vista.mostrarEventos()
        end
        
        if(!@juego.final_del_juego())
          
          if(operacion == Operaciones_juego::COMPRAR)
          
            #respuesta = @vista.comprar()
            respuesta = Respuestas::NO
            if(respuesta == Respuestas::SI)
              
              @juego.comprar()
              
            end
            
            @juego.siguiente_paso_completado(operacion)
          end
          
          if(operacion == Operaciones_juego::GESTIONAR)
          
            @vista.gestionar()
            
            ig = @vista.getGestion
            ip = @vista.getPropiedad

            lista_gestiones_inmobiliarias = [Gestiones_inmobiliarias::VENDER,
              Gestiones_inmobiliarias::HIPOTECAR, Gestiones_inmobiliarias::CANCELAR_HIPOTECA,
              Gestiones_inmobiliarias::CONSTRUIR_CASA, Gestiones_inmobiliarias::CONSTRUIR_HOTEL, Gestiones_inmobiliarias::TERMINAR]
            
            oi = Operacion_inmobiliaria.new(lista_gestiones_inmobiliarias[ig], ip)
                    
            if(oi.gestion == Gestiones_inmobiliarias::CANCELAR_HIPOTECA)
              
              @juego.cancelar_hipoteca(ip)
            
            elsif(oi.gestion == Gestiones_inmobiliarias::CONSTRUIR_CASA)
              
              @juego.construir_casa(ip)
            
            elsif(oi.gestion == Gestiones_inmobiliarias::CONSTRUIR_HOTEL)
              
              @juego.construir_hotel(ip)
              
            elsif(oi.gestion == Gestiones_inmobiliarias::HIPOTECAR)
              
              @juego.hipotecar(ip)
              
            elsif(oi.gestion == Gestiones_inmobiliarias::VENDER)
              
              @juego.vender(ip)
             
            else
              
              @juego.siguiente_paso_completado(operacion)
              
            end
          end
                
          if(operacion == Operaciones_juego::SALIR_CARCEL)
                
            #salida = @vista.salirCarcel()
              salida = Salidas_carcel::PAGANDO      
            if(salida == Salidas_carcel::PAGANDO)
                    
              @juego.salir_carcel_pagando()
                    
            else
                    
              @juego.salir_carcel_tirando()
            end
                   
            @juego.siguiente_paso_completado(operacion)           
          end
        end
      end
        
      ranking = @juego.ranking()
      
      puts("===================================================================")
      puts("Ranking:\n\n")
      
      for i in 0..ranking.length() do
        
        puts(ranking.at(i).to_s + "\n")
        
      end
      
    end
  end
end